src/renderers/WebGLRenderer.js

```diff
@@ -1502,6 +1504,10 @@ function WebGLRenderer( parameters ) {
 
 				initMaterial( material, scene, object );
 
+			} else if ( materialProperties.envMap !== envMap ) {
```

Comment:
Unfortunately, there is no way around this check. However, if an existing cube map is changed with another one, `initMaterial()` will not trigger the expensive shader compilation.